File: src/nikaResponse.ts
export interface NikaResponse {
  VERTICAL_CLASSES: boolean;
  SHOW_TEACHERS: boolean;
  SECOND_RELATIVE: boolean;
  STRIKEOUT_FREE_LSN: boolean;
  SHOW_EXCHANGES_TERM: boolean;
  DISABLE_LINK_LOGO: boolean;
  CLASSES_BTN: string;
  TEACHERS_BTN: string;
  WEEKDAYNUM: number;
  LESSONSINDAY: number;
  FIRSTLESSONNUM: number;
  USEROOMS: boolean;
  SCHOOL_NAME: string;
  CITY_NAME: string;
  EXPORT_DATE: string;
  EXPORT_TIME: string;
  LANG: string;
  DAY_NAMES: string[];
  DAY_NAMESH: string[];
  MONTHS: string[];
  MONTHS2: string[];
  MONTHS3: string[];
  LESSON_NUM_STR: string;
  LESSON_STR: string;
  DAY_NUM_STR: string;
  CLASS_STR: string;
  FOR_CLASS_STR: string;
  SCHEDULE_STR: string;
  FOR_TEACHER_STR: string;
  PERIOD_STR: string;
  SECOND_SHIFT_STR: string;
  TIME_GO: string;
  TIME_START: string;
  TIME_REMAIN: string;
  NO_STR: string;
  NO_LESSONS_STR: string;
  LESSON_CANCELED_STR: string;
  METHOD_STR: string;
  MINUTES1: string;
  MINUTES2: string;
  MINUTES3: string;
  YEAR_STR: string;
  FROM_STR: string;
  FOR_WEEK: string;
  CHANGES_STR: string;
  PREVIOUS_STR: string;
  PREVIOUS2_STR: string;
  NEXT_STR: string;
  NEXT2_STR: string;
  MONTH_STR: string;
  WEEK_STR: string;
  IN_STR: string;
  SCHEDULE2_STR: string;
  REFRESH_STR: string;
  TODAY_STR: string;
  YESTERDAY_STR: string;
  ROOM_FREE_STR: string;
  DAY_NO_LSN_STR: string;
  DAY_NO_PERIOD_STR: string;
  INFO_PANELS: InfoPanels;
  PANEL_COLORS: PanelColors;
  PANEL_PARAMS: PanelParams;
  CLASS_SHIFT: ClassShift;
  TEACHERS: Teachers;
  SUBJECTS: Subjects;
  CLASSES: Classes;
  CLASS_COURSES: ClassCourses;
  ROOMS: Rooms;
  CLASSGROUPS: Classgroups;
  PERIODS: Periods;
  LESSON_TIMES: LessonTimes;
  CLASS_SCHEDULE: ClassSchedule;
  CLASS_EXCHANGE: ClassExchange;
  TEACH_SCHEDULE: TeachSchedule;
  TEACH_EXCHANGE: TeachExchange;
}

export interface InfoPanels {
  "1": InfoPanel[];
}

export interface InfoPanel {
  TYPE: string;
  PARAM: string;
  TITLE: string;
  SCRN_CNT: number;
  SCRN_TIME: number;
  REF: string;
}

export interface PanelColors {
  BACKGRND: string;
  TITLE_FONT: string;
  CLASS_FONT: string;
  DEFAULT_SUBJECT: string;
  EXCHANGE_TEXT: string;
  DEFAULT_ROOM: string;
  CLOCK: string;
  CURR_LSN_FONT: string;
  DEFAULT_ROW1: string;
  DEFAULT_ROW2: string;
  CURR_ROW1: string;
  CURR_ROW2: string;
}

export interface PanelParams {
  AUTO_CHANGE_FONT: boolean;
  SHOW_EXCHANGES: boolean;
  SHOW_CLOCK: boolean;
  SHOW_LSN_TIMES: boolean;
}

export interface ClassShift {
  "96": ClassShift96;
}

export type ClassShift96 = Record<string, number>;

export type Teachers = Record<string, string>;

export type Subjects = Record<string, string>;

export type Classes = Record<string, string>;

export type ClassCourses = Record<string, string>;

export type Rooms = Record<string, string>;

export type Classgroups = Record<string, string>;

export interface Periods {
  "96": Period;
}

export interface Period {
  b: string;
  e: string;
  name: string;
}

export type LessonTimes = Record<string, string[]>;

export interface ClassSchedule {
  "96": ClassSchedule96;
}

export type ClassSchedule96 = Record<string, DaySchedule>;

export type DaySchedule = Record<string, Lesson>;

export interface Lesson {
  s: string[];
  t: string[];
  r: string[];
}

export type ClassExchange = Record<string, Days>;

export type Days = Record<string, Day>;

export type Day = Record<string, DayField>;

export interface DayField {
  s: string[];
  t: string[];
  g: string[];
  r: string[];
}

export interface TeachSchedule {
  "96": TeachSchedule96;
}

export type TeachSchedule96 = Record<string, TeacherTable>;

export type TeacherTable = Record<string, TeacherField>;

export interface TeacherField {
  s: string;
  c: string[];
  r: string;
}

export type TeachExchange = Record<string, TeachExchangeDates>;

export type TeachExchangeDates = Record<string, TeacherExchangeTable>;

export type TeacherExchangeTable = Record<string, TeacherExchangeField>;

export interface TeacherExchangeField {
  s: string;
  c: string[];
  r: string;
}

File: src/nika_response.rs
use std::collections::HashMap;

use serde::Deserialize;

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
#[serde(rename_all(deserialize = "SCREAMING_SNAKE_CASE"))]
pub struct NikaResponse {
    pub vertical_classes: bool,
    pub show_teachers: bool,
    pub second_relative: bool,
    pub strikeout_free_lsn: bool,
    pub show_exchanges_term: bool,
    pub disable_link_logo: bool,
    pub classes_btn: String,
    pub teachers_btn: String,
    pub weekdaynum: i64,
    pub lessonsinday: i64,
    pub firstlessonnum: i64,
    pub userooms: bool,
    pub school_name: String,
    pub city_name: String,
    pub export_date: String,
    pub export_time: String,
    pub lang: String,
    pub day_names: Vec<String>,
    pub day_namesh: Vec<String>,
    pub months: Vec<String>,
    pub months2: Vec<String>,
    pub months3: Vec<String>,
    pub lesson_num_str: String,
    pub lesson_str: String,
    pub day_num_str: String,
    pub class_str: String,
    pub for_class_str: String,
    pub schedule_str: String,
    pub for_teacher_str: String,
    pub period_str: String,
    pub second_shift_str: String,
    pub time_go: String,
    pub time_start: String,
    pub time_remain: String,
    pub no_str: String,
    pub no_lessons_str: String,
    pub lesson_canceled_str: String,
    pub method_str: String,
    pub minutes1: String,
    pub minutes2: String,
    pub minutes3: String,
    pub year_str: String,
    pub from_str: String,
    pub for_week: String,
    pub changes_str: String,
    pub previous_str: String,
    pub previous2_str: String,
    pub next_str: String,
    pub next2_str: String,
    pub month_str: String,
    pub week_str: String,
    pub in_str: String,
    pub schedule2_str: String,
    pub refresh_str: String,
    pub today_str: String,
    pub yesterday_str: String,
    pub room_free_str: String,
    pub day_no_lsn_str: String,
    pub day_no_period_str: String,
    pub info_panels: InfoPanels,
    pub panel_colors: PanelColors,
    pub panel_params: PanelParams,
    pub class_shift: ClassShift,
    pub teachers: Teachers,
    pub subjects: Subjects,
    pub classes: Classes,
    pub class_courses: ClassCourses,
    pub rooms: Rooms,
    pub classgroups: Classgroups,
    pub periods: Periods,
    pub lesson_times: LessonTimes,
    pub class_schedule: ClassScheduleEntry,
    pub class_exchange: ClassExchange,
    pub teach_schedule: TeachSchedule,
    pub teach_exchange: TeachExchange,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct InfoPanels(pub HashMap<String, InfoPanel>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
#[serde(rename_all(deserialize = "SCREAMING_SNAKE_CASE"))]
pub struct InfoPanel {
    pub type_field: String,
    pub param: String,
    pub title: String,
    pub scrn_cnt: i64,
    pub scrn_time: i64,
    pub ref_field: String,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
#[serde(rename_all(deserialize = "SCREAMING_SNAKE_CASE"))]
pub struct PanelColors {
    pub backgrnd: String,
    pub title_font: String,
    pub class_font: String,
    pub default_subject: String,
    pub exchange_text: String,
    pub default_room: String,
    pub clock: String,
    pub curr_lsn_font: String,
    pub default_row1: String,
    pub default_row2: String,
    pub curr_row1: String,
    pub curr_row2: String,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
#[serde(rename_all(deserialize = "SCREAMING_SNAKE_CASE"))]
pub struct PanelParams {
    pub auto_change_font: bool,
    pub show_exchanges: bool,
    pub show_clock: bool,
    pub show_lsn_times: bool,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassShifts(pub HashMap<String, ClassShift>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassShift(pub HashMap<String, i64>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Teachers(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Subjects(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Classes(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassCourses(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Rooms(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Classgroups(pub HashMap<String, String>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Periods(pub HashMap<String, Period>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct Period {
    pub b: String,
    pub e: String,
    pub name: String,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct LessonTimes(pub HashMap<String, Vec<String>>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassSchedule(pub HashMap<String, ClassSchedule1>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassSchedule1(pub HashMap<String, ClassScheduleEntry>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassScheduleEntry {
    pub s: Vec<String>,
    pub t: Vec<String>,
    pub r: Vec<String>,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassExchange(pub HashMap<String, ClassExchange1>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassExchange1(pub HashMap<String, ClassExchange2>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassExchange2(pub HashMap<String, ClassExchangeEntry>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct ClassExchangeEntry {
    pub s: Vec<String>,
    pub t: Vec<String>,
    pub r: Vec<String>,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachSchedule(pub HashMap<String, TeachSchedule1>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachSchedule1(pub HashMap<String, TeachSchedule2>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachSchedule2(pub HashMap<String, TeachScheduleEntry>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachScheduleEntry {
    pub s: String,
    pub c: Vec<String>,
    pub r: String,
}

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachExchange(pub HashMap<String, TeachExchange1>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachExchange1(pub HashMap<String, TeachExchange2>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachExchange2(pub HashMap<String, TeachExchangeEntry>);

#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
pub struct TeachExchangeEntry {
    pub s: String,
    pub c: Vec<String>,
    pub r: String,
}
